### PR TITLE
All features requested were developed and project is functional now

### PR DESCRIPTION
**Pull request by cochabambinoski**
The project was completed, which include:
- Users name registration to play
- Reset button
- Functional logic for the game as itself

### CODE REVIEW

File: js/script.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+      if (playerForm.name1.value > '' && playerForm.name2.value > '') {
+        players[0] = playerForm.name1.value;
+        players[1] = playerForm.name2.value;
+        btn.disabled = false;
+      } else {
+        btn.disabled = true;
+      }
+    });
+  });
+};
```
Comment (shloch) [inline js/script.js]: Separate Game logic from DOM manipulation [[see requirements](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/section/dom_logic_separation_requirement.png)]

**Review by shloch — CHANGES_REQUESTED** [review]
## hello @cochabambinoski , @davitomix 
Nice attempt to solve this game, and you guys successfully solved it
It works perfectly
Screen Shot 2020-04-08 at 6 51 29 PM

Screen Shot 2020-04-08 at 6 51 43 PM


### [required changes]
Unfortunately you guys did not quite respect the guidelines........
You were supposed to follow the guidelines reccommended by  [THE ODIN PROJECT](https://www.theodinproject.com/courses/javascript/lessons/tic-tac-toe-javascript), as stated by [MICROVERSE PATHWRIGHT](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59623090/)

- the goal of this project is to understand and use javascript modules and factories. You therefore have to encapsulate/wrap your code around this philosophy. It's weird in begining, but you'll get it, makes code organisation easier and DEBUGGING

- Also You needed to separate LOGIC from DOM

### we're have to get back to debugging to fit this reality.......sorry about that guys

It's a good thing, your code works, so you have some refactoring to do

### happy coding

**Comment by shloch** [comment]
You're not limited to using ONE javascript files to organise your code.... You can for example put LOGIN in one file and DOM in another js file,

**Comment by davidmrtz-dev** [comment]
- Refactorized the entire program as required by TSE.
- Encapsulated functions following and using factory functions.
- Added UI to let the users enter their names. 
- Fixed and checked lint issues.

**Review by mcrd25 — CHANGES_REQUESTED** [review]
# Changes Required 🔍 👀 

Hi @cochabambinoski and @davitomix! 👋 

As mentioned before, great job with game! However, there still needs to be separation of application logic from DOM manipulation.

A hint of how t accomplish this:
As hinted from this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/tic_tac_toe/objects_requirement.png) from Odin. You ought to have three objects: Gameboard, Player, and Game. I suggest you contain the application logic in these objects and manipulate the Dom based on calls from these objects. I also suggest you put each object in separate files and leave DOM manipulation in the script file.

For instance, your player object would be a factory function like the following:
```js
const PlayerFactory = (name, symbol) => ({ name, symbol });
export default PlayerFactory;
```
Your Board object could be a module pattern some thing like this:
```js
const Board = (() => {
  let board = Array(9).fill(null);

  const returnBoard = () => board;

  const getCell = (index) => board[index];

  
    // Maybe a setCell method etc.


  const reset = () => {
    board = Array(9).fill(null);
  };

  // more functions

  const winningPositions = () => {
    // winning positions logic based on board array
    return positions;
  };

  return {
    returnBoard,
    getCell,
    // other functions that would be used publicly,
    winningPositions,
    reset,
  };
})();

export default Board;
```
Above is just an example, you can use this as a guide. Leave the application logic in this module.

If you want to create an object for DOM manipulation, make it another object like `BoardDom` etc.

Your `Game` object could also be a module pattern.

This is, of course, all a guide. You can implement your `Game`, `Board` and `Player` objects however you like.   Just make sure to separate application logic from DOM manipulation.

Hope this helps you meet the objects and DOM manipulation/ Application Logic separation.

***Please makes changes and submit another review request.
If you have any questions, let me know here or on slack and I will be glad to clarify with you. You can also reach me and connect with me on my social media or email.***

*Happy coding 😄,*
**[--Maya](https://github.com/mcrd25)**
[GitHub](https://github.com/mcrd25) [Twitter](https://twitter.com/mcrd25) [linkedin](https://www.linkedin.com/in/mayadouglas/) [GitLab](https://gitlab.com/mcrd25)

---
*P.S. all requested changes not marked by [OPTIONAL] or are not positive comments are **REQUIRED***

**Comment by davidmrtz-dev** [comment]
Hi,
Thanks for your time to review our project,
We already added the requested changes by the previous TSE's.

- Separate DOM manipulation/ Application Logi.
- Added modules and use factory functions to make the program.
- Implemented module pattern.

So to make this repo work you have to open live server VSC Code, because if you open the file directly on the browser the console return something like this and the program doesn't work: 

'Access to script at 'file:///Volumes/Macintosh%20External%20Drive/mv_js/titactoe/Tic-Tac-Toe-Project/js/main.js' from origin 'null' has been blocked by CORS policy: Cross origin requests are only supported for protocol schemes: http, data, chrome, chrome-extension, https.' 

Searching on google I found this: [ES6 modules are subject to same-origin policy](https://stackoverflow.com/questions/52919331/access-to-script-at-from-origin-null-has-been-blocked-by-cors-policy) so I found this is an error of the browsers like says this post ➡️ [CORS](https://stackoverflow.com/questions/46992463/es6-module-support-in-chrome-62-chrome-canary-64-does-not-work-locally-cors-er?rq=1)

So if you run the program on a live server it runs very well.

Thank you.

**Review by Oghenebrume50 — APPROVED** [review]
### STATUS: APPROVED 

Hi Team great job :clap:  nice code and kudos for following through with all the previous suggested corrections, everything looks okay now :+1: please merge and move on, **this is approved :slightly_smiling_face:**  

**RAPHAEL NORIODE** :world_map:

**Pull request merged**